python/ray/ml/predictors/integrations/torch/torch_predictor.py

```diff
@@ -0,0 +1,67 @@
+from typing import Optional, Union, List
+
+import torch
+
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+
+
+class TorchPredictor(Predictor):
+    """A predictor for PyTorch models.
+
+    Args:
+        model: The torch module to use for predictions.
+        preprocessor: The preprocessor used to transform data batches prior
+            to prediction.
+
+    """
+
+    def __init__(self, model: torch.nn.Module, preprocessor: Preprocessor):
+        raise NotImplementedError
+
+    @classmethod
+    def from_checkpoint(
+        cls, checkpoint: Checkpoint, model_definition: Optional[torch.nn.Module] = None
+    ) -> "TorchPredictor":
+        """Instantiate the predictor from a Checkpoint.
+
+        The checkpoint is expected to be a result of ``TorchTrainer``.
+
+        Args:
+            checkpoint: The checkpoint to load the model and
+                preprocessor from. It is expected to be from the result of a
+                ``TorchTrainer`` run.
+            model_definition: If checkpoints
```

Comment:
I think this needs to be clearer in terms of what `model_definition` actually is. Both `model` and this one are `torch.nn.Module`s and I'm not sure users will understand what's in our checkpoint and what's not.

Reply:
+1

Reply:
Ok changed the name to `model`, does that help?

Also it would help if you guys could clarify on what's not clear. What's missing from the docstring in your mind?

Regarding whether users will understand what's in our checkpoint, they are the ones who are creating the checkpoint, so they should be aware of this.